### PR TITLE
8297489: Modify TextAreaTextEventTest.java as to verify the content change of TextComponent sends TextEvent

### PR DESCRIPTION
**Pull request by ravigupta30**
Modify TextAreaTextEventTest.java as to verify the content changes of a both TextComponent(TextField,TextArea) for the following assertions.

a. TextListener get invoked when the content of a TextField gets changed.
b. TextListener not get invoked during text selection or when Special keys such as Function Keys are pressed.

Testing:
Tested using Mach5(20 times per platform) in macos,linux and windows and got all pass.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8297489](https://bugs.openjdk.org/browse/JDK-8297489): Modify TextAreaTextEventTest.java as to verify the content change of TextComponent sends TextEvent

### Reviewers
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - Committer)
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11326/head:pull/11326` \
`$ git checkout pull/11326`

Update a local copy of the PR: \
`$ git checkout pull/11326` \
`$ git pull https://git.openjdk.org/jdk pull/11326/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11326`

View PR using the GUI difftool: \
`$ git pr show -t 11326`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11326.diff">https://git.openjdk.org/jdk/pull/11326.diff</a>

</details>

### CODE REVIEW

**Comment by ravigupta30** [comment]
Any volunteers for a review?

File: test/jdk/java/awt/event/ComponentEvent/TextFieldTextEventTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2007, 2022, Oracle and/or its affiliates. All rights reserved.
```
Comment (honkar-jdk) [inline test/jdk/java/awt/event/ComponentEvent/TextFieldTextEventTest.java:2]: If this is a newly created test, then the copyright should contain only the present year.

File: test/jdk/java/awt/event/ComponentEvent/TextFieldTextEventTest.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
```
Comment (honkar-jdk) [inline test/jdk/java/awt/event/ComponentEvent/TextFieldTextEventTest.java:22]: Minor nit: Please add a newline between copyright header and import statements.

File: test/jdk/java/awt/event/ComponentEvent/TextFieldTextEventTest.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        }
+    }
+
+    public static void disposeFrame() {
+        if (frame != null) {
+            frame.dispose();
+        }
+    }
+
+    private static void typeKey(int key) throws Exception {
```
Comment (honkar-jdk) [inline test/jdk/java/awt/event/ComponentEvent/TextFieldTextEventTest.java:136]: `throws Exception` clause can be removed.

File: test/jdk/java/awt/event/ComponentEvent/TextFieldTextEventTest.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        robot.keyPress(key);
+        robot.keyRelease(key);
+    }
+
```
Comment (honkar-jdk) [inline test/jdk/java/awt/event/ComponentEvent/TextFieldTextEventTest.java:140]: Minor nit: extra newline before closing brace can be removed.

**Review by honkar-jdk — CHANGES_REQUESTED** [review]
Tested on Mac 12 and Win10 locally, test works as expected. CI testing looks good.
Minor changes suggested.

File: test/jdk/java/awt/event/ComponentEvent/TextFieldTextEventTest.java (hunk 1 above)
Comment (ravigupta30) [reply test/jdk/java/awt/event/ComponentEvent/TextFieldTextEventTest.java:2]: I have added the original date based on the above comment that’s we get during the PR review below
https://github.com/openjdk/jdk/pull/7512#issuecomment-1075171066

**Review by honkar-jdk — APPROVED** [review]

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]
This test is very similar to that in #11052: `TextField` vs `TextArea` ([JDK-8296632](https://bugs.openjdk.org/browse/JDK-8296632)).

Is it possible to modify the test so that it runs twice for both components?

**Comment by honkar-jdk** [comment]
> This test is very similar to that in #11052: `TextField` vs `TextArea` ([JDK-8296632](https://bugs.openjdk.org/browse/JDK-8296632)).
> 
> Is it possible to modify the test so that it runs twice for both components?

I agree with @aivanov-jdk. The test can be modified to check both TextArea and TextField to avoid code repetition.

**Comment by ravigupta30** [comment]
i am combining the tests  https://github.com/openjdk/jdk/pull/11052: TextField with TextArea ([JDK-8296632](https://bugs.openjdk.org/browse/JDK-8296632)) so that  it runs twice for both components.

**Comment by aivanov-jdk** [comment]
> i am combining the tests #11052: TextField with TextArea ([JDK-8296632](https://bugs.openjdk.org/browse/JDK-8296632)) so that it runs twice for both components.

You can do this under this PR.

Eventually, you're adding another test case to an existing one.

File: test/jdk/java/awt/event/ComponentEvent/TextComponentTextEventTest.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                typeKey(KeyEvent.VK_T);
+
+                textChanged = false;
+                typeKey(KeyEvent.VK_ENTER);
+
+                robot.waitForIdle();
+                if (textComp instanceof TextField && textChanged) {
+                    throw new RuntimeException(
+                        "FAIL: TextEvent triggered when Enter pressed on in "
+                            + textComp);
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/event/ComponentEvent/TextComponentTextEventTest.java:116]:
```suggestion
                        "FAIL: TextEvent triggered when Enter pressed on "
                            + textComp);
```

File: test/jdk/java/awt/event/ComponentEvent/TextComponentTextEventTest.java [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+            robot = new Robot();
+            robot.setAutoDelay(100);
+            robot.setAutoWaitForIdle(true);
+
+            for (TextComponent textComp : components) {
+
+                robot.waitForIdle();
+                EventQueue.invokeAndWait(() -> {
+                    textFieldAt = textComp.getLocationOnScreen();
+                    textFieldSize = textComp.getSize();
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/event/ComponentEvent/TextComponentTextEventTest.java:88]:
```suggestion
                    textCompAt = textComp.getLocationOnScreen();
                    textCompSize = textComp.getSize();
```
This makes it clearer that the location and size are of a generic component which could be either `textField` or `textArea`.

File: test/jdk/java/awt/event/ComponentEvent/TextComponentTextEventTest.java [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+        frame.setLayout(new FlowLayout());
+        TextField textField = new TextField(20);
+        textField.addTextListener((event) -> {
+            textChanged = true;
+            System.out.println("TextField got a text event: " + event);
+        });
+        frame.add(textField);
+        TextArea textArea = new TextArea(5, 15);
+        textArea.addTextListener((event) -> {
+            System.out.println("TextArea Got a text event: " + event);
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/event/ComponentEvent/TextComponentTextEventTest.java:63]: Nit for being consistent:
```suggestion
            System.out.println("TextArea got a text event: " + event);
```

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

File: test/jdk/java/awt/event/ComponentEvent/TextComponentTextEventTest.java [hunk 8]
```diff
@@ -112,21 +110,20 @@ public static void main(String[] args) throws Exception {
                 robot.waitForIdle();
                 if (textComp instanceof TextField && textChanged) {
                     throw new RuntimeException(
-                        "FAIL: TextEvent triggered when Enter pressed on in "
-                            + textComp);
+                        "FAIL: TextEvent triggered when Enter pressed on "+ textComp);
```
Comment (aivanov-jdk) [inline test/jdk/java/awt/event/ComponentEvent/TextComponentTextEventTest.java:113]:
```suggestion
                        "FAIL: TextEvent triggered when Enter pressed on " + textComp);
```

**Review by aivanov-jdk — APPROVED** [review]

File: test/jdk/java/awt/event/ComponentEvent/TextComponentTextEventTest.java (hunk 8 above)
Comment (aivanov-jdk) [reply test/jdk/java/awt/event/ComponentEvent/TextComponentTextEventTest.java:115]: It's not resolved: there's still no space before `+` operator.

And you removed it from other places for some reason where they were present.

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

**Review by aivanov-jdk — APPROVED** [review]

**Review by honkar-jdk — APPROVED** [review]

**Pull request closed without merging**